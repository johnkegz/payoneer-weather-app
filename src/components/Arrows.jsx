import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ForwardIcon from '@material-ui/icons/Forward';

const useStyles = makeStyles({
  container: {
    display: 'flex',
    alignItems: 'center',
    cursor: 'pointer'
   },
   hideArrow: {
     display: 'none'
   }
});


export function RightArrow({next, end, data}) {
    const classes = useStyles();
    return <div 
      className={`${classes.container} ${end===data.length?classes.hideArrow: {}}`} 
      onClick={() => next()}
    >
      <ForwardIcon style={{ color: 'rgb(66, 133, 244)', fontSize: '100px',}}/>
    </div>;
}

export function LeftArrow({prev, start}) {
  const classes = useStyles();
  return <div 
      className={`${classes.container} ${start===0?classes.hideArrow: {}}`} 
      onClick={() => prev()}
    >
      <ForwardIcon style={{transform: 'rotate(180deg)', color: 'rgb(66, 133, 244)', fontSize: '100px'}}/>
  </div>;
}
