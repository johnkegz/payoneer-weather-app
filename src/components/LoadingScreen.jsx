import React from 'react';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
    container: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '100vh',
        fontSize: '1.25rem',
        width: '100%'
     },
     
  });

function LoadingScreen() {
    const classes = useStyles();
    return (
        <div className={classes.container}>
            loading....
        </div>
    )
}

export default LoadingScreen;

