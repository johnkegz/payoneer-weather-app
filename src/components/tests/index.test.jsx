import React from "react";
import { shallow } from "enzyme";
import LoadingScreen from "../LoadingScreen";
import { RightArrow, LeftArrow } from "../Arrows";
import WearherCard from "../WeatherCard";

describe("Loading screen", () => {
  let LoadingScreenWrapper;
  let RightArrowScreenWrapper;
  let LeftArrowWrapper;
  let WearherCardWrapper;
  beforeEach(() => {
    LoadingScreenWrapper = shallow(<LoadingScreen />);
    RightArrowScreenWrapper = shallow(<RightArrow end={1} data={[{id: 1}]} />);
    LeftArrowWrapper = shallow(<LeftArrow start={0} />);
    WearherCardWrapper = shallow(<WearherCard weather={{rawDate: new Date(), date: new Date(), weatherSegments: [{dt: new Date(), main: {temp: 278}}]}} currentWeather={{date: new Date()}} />);
  });

  it("Loading screen has one div", () => {
    expect(LoadingScreenWrapper.find("div").length).toEqual(1);
  });
  it("Right arrow has one div", () => {
    expect(RightArrowScreenWrapper.find("div").length).toEqual(1);
  });
  it("Left arrow has one div", () => {
    expect(LeftArrowWrapper.find("div").length).toEqual(1);
  });
  it("Render WearherCardWrapper", () => {
    expect(WearherCardWrapper.find("h2")).toEqual({});
  });
});
