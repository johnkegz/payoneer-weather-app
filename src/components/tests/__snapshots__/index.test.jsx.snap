// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading screen Weather information screen to match snapshot 1`] = `
<LoadingScreen>
  <div
    className="makeStyles-container-1"
  >
    loading....
  </div>
</LoadingScreen>
`;
